collections_hierarchy/mixins.py

```diff
@@ -1,87 +1,98 @@
 class ComparableMixin(object):
     def __eq__(self, other):
-        pass
+        return self == other
 
     def __ne__(self, other):
         return not self == other
 
     def __lt__(self, other):
-        pass
+        return len(self) < len(other)
 
     def __gt__(self, other):
-        pass
+        return len(self) > len(other)
 
     def __le__(self, other):
-        pass
+        return len(self) <= len(other)
 
     def __ge__(self, other):
-        pass
+        return len(self) >= len(other)
 
 
 class SequenceMixin(object):
     def __iter__(self):
-        pass
+        return iter(self)
```

Comment:
I think this is an infinite recursion since iter(self) will call iter(self) etc. Instead you could just return self and have next do the step-through iteration. Another suggestion is that instead of initializing the counter variable i in the class inits you could initialize it here in iter. If you only initialize it in the class init it means that you can only do a single iteration because i won't be reset if you call a for a second iteration.